### ISSUE TITLE
Hanging on extraction?

### ISSUE THREAD

**Issue by travisboss** [davidnewhall/unpackerr#46]
```
2020/09/02 11:24:00.175079 pollers.go:121: [DEBUG] Radarr: Status: Batman.Begins.2005.1080p.BluRay.REMUX.VC-1.TrueHD.5.1-Kanibus@PrivateHD.mkv (Waiting, pre-Queue, elapsed: 2h16m0s, found: true)
2020/09/02 11:24:00.175150 pollers.go:121: [DEBUG] Radarr: Status: Caddyshack.1980.BluRay.1080p.DTS-HD.MA.5.1.VC-1.REMUX-FraMeSToR (Waiting, pre-Queue, elapsed: 2h16m0s, found: true)
2020/09/02 11:24:00.175193 pollers.go:121: [DEBUG] Sonarr: Status: Goliath.S02E08.INTERNAL.720p.HEVC.x265-MeGusta (Waiting, pre-Queue, elapsed: 2h16m0s, found: true)
2020/09/02 11:24:00.175268 pollers.go:121: [DEBUG] Sonarr: Status: Goliath.S02E05.INTERNAL.REPACK.720p.HEVC.x265-MeGusta (Waiting, pre-Queue, elapsed: 2h16m0s, found: true)
2020/09/02 11:24:00.175308 pollers.go:121: [DEBUG] Sonarr: Status: Goliath.S02E06.INTERNAL.REPACK.720p.HEVC.x265-MeGusta (Waiting, pre-Queue, elapsed: 2h16m0s, found: true)
2020/09/02 11:24:00.175400 pollers.go:121: [DEBUG] Sonarr: Status: The.Grand.Tour.S03E02.REPACK.720p.WEB.H264-METCON (Extracted, Awaiting Import, elapsed: 2h11m40s, found: true)
2020/09/02 11:24:00.175443 pollers.go:121: [DEBUG] Sonarr: Status: Goliath.S02E07.INTERNAL.720p.HEVC.x265-MeGusta (Waiting, pre-Queue, elapsed: 2h16m0s, found: true)
2020/09/02 11:24:00.175532 pollers.go:121: [DEBUG] Sonarr: Status: Goliath.S02E03.INTERNAL.720p.HEVC.x265-MeGusta (Waiting, pre-Queue, elapsed: 2h16m0s, found: true)
2020/09/02 11:24:00.175700 pollers.go:121: [DEBUG] Radarr: Status: 500.Days.of.Summer.1080p.2009.DTS-HD.MA.5.1.AVC.REMUX-FraMeSToR (Waiting, pre-Queue, elapsed: 2h16m0s, found: true)
2020/09/02 11:24:00.175843 logs.go:57: [Unpackerr] Queue: [8 waiting] [0 queued] [0 extracting] [1 extracted] [0 imported] [0 failed] [0 deleted], Totals: [0 restarts] [0 finished]
2020/09/02 11:25:00.013613 logs.go:57: [Unpackerr] Queue: [8 waiting] [0 queued] [0 extracting] [1 extracted] [0 imported] [0 failed] [0 deleted], Totals: [0 restarts] [0 finished]
```

New issue here, not sure how to resolve this one, but I am guessing with some left over `rar` files it is trying to extract but sonarr/radarr already have these files?  Can I have it not try to extract or is there a config piece I am missing?

```
 Unpackerr Example Configuration File

# [true/false] Turn on debug messages in the output. Do not wrap this in quotes.
# Recommend trying this so you know what it looks like. I personally leave it on.
debug = true

# Disable writing messages to stdout. This silences the app. You should set a log
# file below if you set this to true.
quiet = false

# Write message to a log file. This is the same data that is normally output to stdout.
# This setting is great for Docker users that want to export their logs to a file.
# The alternative is to use syslog to log the output of the application to a file.
# Default is no log file, so this is unset.
#log_file = "/downloads/unpackerr.log"

# How often to poll sonarr and radarr.
# Recommend 1m-5m. Uses Go Duration.
interval = "2m"

# How long to wait for a reply from the backends: Sonarr, Radarr, Lidarr.
# This can also be set per-app. Uses Go Duration.
timeout = "1m"

# How long ago a file must have been imported before deletion. The file must
# also not be part of an active queue item. Set this to "1m" to make sure files
# are deleted quickly after being imported. Recommend "5m". Uses Go Duration.
delete_delay = "5m"

# How long an item must be queued (download complete) before extraction will start.
# One minute is the historic default and works well. Set higher if your downloads
# take longer to finalize (or transfer locally). Uses Go Duration.
start_delay = "10s"

# How long to wait before removing the history for a failed extraction.
# Once the history is deleted the item will be recognized as new and
# extraction will start again. Uses Go Duration.
retry_delay = "5m"

# How many files may be extracted in parallel. 1 works fine.
# Do not wrap the number in quotes. Raise this only if you have fast disks and CPU.
parallel = 2

# The following sections can be repeated if you have more than one Sonarr, Radarr or Lidarr.
# You must uncomment the [[header]] and api_key at a minimum.

[[sonarr]]
   url = "http://sonarr:8989"
   api_key = "hiddenkey"
# File system path where downloaded Sonarr items are located.
   path = "/downloads/sonarr"

[[radarr]]
  url = "http://radarr:7878"
  api_key = "hiddenkey"
# File system path where downloaded Radarr items are located.
  path = "/downalods/radarr"
```
here is my config, pretty standard as I understand it.

```
  unpackerr:
    image: golift/unpackerr
    container_name: unpackerr
    environment:
      - TZ=${TZ}
      - PUID=${PUID}
      - PGID=${PGID}
    volumes:
      - ./unpackerr:/etc/unpackerr/
      - /mnt/torrent/complete:/downloads
    restart: unless-stopped
````
compose config

The way my seedbox is setup is downloading files are in `incoming` once finished are moved to `completed/sonarr or radarr` where they are picked up by there respective applications and imported.

**Comment by davidnewhall** [comment]
A couple questions to start:
- Does anything ever get "imported" after it's extracted?
- Do you have this same mount path in your download client container? `- /mnt/torrent/complete:/downloads`
- The logs you provided do not list the two "extracted" items. Do those show up in the logs?
- Can you provide a larger log snippet? Putting it in a [gist](https://gist.github.com) is OK. I feel like there should be more valuable information in there.
- Can you describe the problem? You asked some question in your description, but I need to make sure I understand what you're seeing. Based on the logs you provided it looks like two different things are happening:
   - Some files are not being marked as finished and thus not being extracted.
   - The items that were extracted are not being imported. **What does Radarr/Sonarr say about these?**

**Comment by travisboss** [comment]
1. These have been previously imported form a previous extraction so it is extracting what was done long ago.
2. No, the page is `/mnt/torrent:/downloads` then rtorrent moves it to `complete/sonarr or /radarr`
    a. changed this back in the config to just `/downloads`
3. They do show as extracted.
4. https://gist.github.com/travisboss/006fb40e472c77cff81484c46eeda807
   a. If you need more logs I'll just leave it going overnight and see what I get.  

The problem is the 9 items just sit in waiting or one or two get extracted but seem to just sit that way, sits like this for hours.

These items have already been imported long ago, just setting up unpackerr for the first time so I do not need to do it manually, but I am still seeding these items.

**Comment by davidnewhall** [comment]
Okay, these lines:
```
2020/09/03 16:10:22.610293 pollers.go:121: [DEBUG] Sonarr: Status: Move.Title (Waiting, pre-Queue, elapsed: 2m0s, found: true)
```
-- there's about 9 of them. This indicates an item is in the Sonarr (or Radarr) queue. If it's been imported... why is it in the queue? Unpackerr is basically "waiting" for it to do something. Either finish or import. If it imports, it drops out of the queue. If it finishes, unpackerr begins working on it. Thoughts?

**Comment by travisboss** [comment]
@davidnewhall Sorry just saw this now, I will try to do an import manually to see if that drops them out of the queue.  I will join the discord server too.

**Comment by travisboss** [comment]
I think I found the issue and it was waiting on an import, rutorrents extraction is I have found horribly unreliable so I had to manually extract some rars, but this does not show up in sonarr/radarr, I am manually doing imports now and the queue is going away!  Thanks @davidnewhall
